invenio_records_resources/services/records/facets/facets.py

```diff
@@ -0,0 +1,250 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Facets types defined."""
+
+from elasticsearch_dsl import A, Q
+from elasticsearch_dsl import TermsFacet as TermsFacetBase
+
+
+class LabelledFacetMixin:
+    """Mixin class for overwriting the default get_values() method."""
+
+    def __init__(self, label=None, value_labels=None, **kwargs):
+        """Initialize class."""
+        self._label = label or ''
+        self._value_labels = value_labels
+        super().__init__(**kwargs)
+
+    def get_value(self, bucket):
+        """Get key value for a bucket."""
+        return getattr(bucket, "key_as_string", bucket.key)
```

Comment:
why `key_as_string` and not `label`? it's the key it's looking for?

Reply:
This is because we parse all facets as strings. E.g. ``/api/records?is_published=1`` ends up as ``{'is_published': ['1']}`` when parsed. The string is then sent to Elasticsearch as a term query on a boolean field, and then Elasticsearch complains.

Honestly, the right solution here is likely that facets url arguments are explicitly parsed and validated for a specific type. I had to do above, in order to get the Status facet to work, and is equivalent to what we did before.

Before, we output the ``key_as_string`` in the aggregations, and RSK would use that key to query with. Now, I just moved this logic down into the service, so that we stay with having only one ``key`` field in the aggregations.